Use PTEFile class in serialize_pte_binary

lucylq opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #15801
* #15800

Take in a PTE file for serialization instead of program, mutable_segments, named_data segments separately.

Differential Revision: [D86908241](https://our.internmc.facebook.com/intern/diff/D86908241/)

lucylq committed:
Use PTEFile class in serialize_pte_binary

Take in a PTE file for serialization instead of program, mutable_segments, named_data segments separately.

Differential Revision: [D86908241](https://our.internmc.facebook.com/intern/diff/D86908241/)

[ghstack-poisoned]

lucylq committed:
Update on "Use PTEFile class in serialize_pte_binary"


Take in a PTE file for serialization instead of program, mutable_segments, named_data segments separately.

Differential Revision: [D86908241](https://our.internmc.facebook.com/intern/diff/D86908241/)

[ghstack-poisoned]

larryliu0820 commented on devtools/bundled_program/test/test_bundle_data.py:
```diff
@@ -18,7 +18,7 @@
     get_common_executorch_program,
 )
-from executorch.exir._serialize import _serialize_pte_binary
+from executorch.exir._serialize import _PTEFile, _serialize_pte_binary
```
Is `PTEFile` supposed to be private only?

lucylq replied:
Mostly following _serialize_pte_binary and _deserialize_pte_binary ...

I think those are fairly public though haha

lucylq committed:
Update on "Use PTEFile class in serialize_pte_binary"


Take in a PTE file for serialization instead of program, mutable_segments, named_data segments separately.

Differential Revision: [D86908241](https://our.internmc.facebook.com/intern/diff/D86908241/)

[ghstack-poisoned]

lucylq committed:
Update on "Use PTEFile class in serialize_pte_binary"


Take in a PTE file for serialization instead of program, mutable_segments, named_data segments separately.

Differential Revision: [D86908241](https://our.internmc.facebook.com/intern/diff/D86908241/)

[ghstack-poisoned]

lucylq committed:
Update on "Use PTEFile class in serialize_pte_binary"


Take in a PTE file for serialization instead of program, mutable_segments, named_data segments separately.

Differential Revision: [D86908241](https://our.internmc.facebook.com/intern/diff/D86908241/)

[ghstack-poisoned]

lucylq committed:
Update on "Use PTEFile class in serialize_pte_binary"


Take in a PTE file for serialization instead of program, mutable_segments, named_data segments separately.

Differential Revision: [D86908241](https://our.internmc.facebook.com/intern/diff/D86908241/)

[ghstack-poisoned]
